src/messages/data.ts

```diff
@@ -0,0 +1,46 @@
+import { base64url } from 'multiformats/bases/base64';
+import { CID } from 'multiformats/cid';
+import { importer } from 'ipfs-unixfs-importer';
+
+export class Data {
+  data: string | number | boolean | Uint8Array | object;
+
+  constructor(data: any) {
+    if (!data) {
+      throw new Error('data cannot be null');
+    }
+
+    this.data = data;
+  }
+
+  toBytes(): Uint8Array {
```

Comment:
I kinda wanna just stick with convertible raw types (e.g. `Uint8Array`), 'coz if I understand current implementation correctly, the type info is lost the moment we serialize and store the bytes in DB. Else the dev experience is kinda half baked and they'll have to either read this method to understand what to expect, or make assumptions when data is returned to them as bytes upon querying.

Reply:
Nixed the data class. relevant functions now live within the `utils` directory. We can accept a wide range of types for `data` just for the static `create` methods to make it easier for end-users. that's the hope at least

Reply:
Got it, just making sure there is no disconnect here: we discussed that the job is only half way and unless we also deserialize back to the original form right? My understanding is that we will do this by setting the optional IPFS mime type property thingy, which I am okay for us to now further bloat the PR with, perhaps we can create an issue and reference it here with a `TODO`?

Reply:
sure i can do that once we get this PR merged. adding a link to this comment into my notes